### PR TITLE
allow a character override in TC Builds

### LINKED ISSUE #2154: Add a "character editor" to TC Build editor
Likely will fulfill https://github.com/frzyc/genshin-optimizer/issues/2144
For people who theorycraft different builds under different conditions, it makes sense to assume a specific character level/talent/constellation. 

Workbreakdown:
- Add character data to TC builds
- Add Character editor to TC editor/overview
- Add additional logic for useTeamData for overriding the character data from buildTC
- Add comparable fields (level/talent/constellation) to show on build results when comparing two builds with different character data
- Likely have to account for comparing results of nodes that may or may not show in comparable builds (like nodes hidden behind constellation)
- Add character data to TC builds when converting build to TC builds

### PR DESCRIPTION
**Pull request by frzyc**
## Describe your changes

Add a way to override character base stats (level, constellation, talents) in TC Builds.
Add extra output fields when comparing between builds of different character base stats

## Issue or discord link

- Resolves #2154

## Testing/validation

Condense the character editor, add a toggle to override the base character 
image

Additional Stat displays to show difference in character base stats
image

## Checklist before requesting a review (leave this PR as draft if any part of this list is not done.)

- [ ] I have commented my code in hard-to understand areas.
- [ ] I have made corresponding changes to README or wiki.
- [ ] For front-end changes, I have updated the corresponding English translations.
- [ ] I have run `yarn run mini-ci` locally to validate format and lint.
- [ ] If I have added a new library or app, I have updated the deployment scripts to ignore changes as needed

### CODE REVIEW

File: libs/gi/ui/src/util/getDisplayUtil.tsx [hunk 1]
```diff
@@ -114,6 +115,7 @@ export function getDisplaySections(data: UIData): DisplaySections {
   )
   return [
     ...basic,
+    ['character', data.getCharacterStats()],
```
Comment (nguyentvan7) [inline libs/gi/ui/src/util/getDisplayUtil.tsx:118]: Character stat comparison showing % increase seems a bit silly

Comment (nguyentvan7) [inline libs/gi/ui/src/util/getDisplayUtil.tsx:118]: Character header still shows even if nothing is there (like when compare is unselected)
image

File: libs/gi/db/src/Interfaces/BuildTc.ts [hunk 2]
```diff
@@ -17,6 +18,7 @@ export type BuildTcArtifactSlot = {
 export type BuildTc = {
   name: string
   description: string
+  character?: Omit<ICharacter, 'key'>
```
Comment (nguyentvan7) [inline libs/gi/db/src/Interfaces/BuildTc.ts:21]: No visualization of override here 
image

Comment (nguyentvan7) [inline libs/gi/db/src/Interfaces/BuildTc.ts:21]: More minor, but also no golden numbers here to show these are overridden values
image

File: libs/gi/ui/src/util/getDisplayUtil.tsx (hunk 1 above)
Comment (frzyc) [reply libs/gi/ui/src/util/getDisplayUtil.tsx:118]: Yeah I had an attempt to fix it: https://github.com/frzyc/genshin-optimizer/blob/c48afc3c4e2db91e074a30cc53bca773511dad73/libs/gi/ui/src/components/character/StatDisplayComponent.tsx#L116

But its actually a lot more complicated to execute

Comment (frzyc) [reply libs/gi/ui/src/util/getDisplayUtil.tsx:118]: I wonder if it makes sense not to show % increase for values below 100?

File: libs/gi/ui/src/components/character/StatDisplayComponent.tsx [hunk 3]
```diff
@@ -42,28 +67,54 @@ export function StatDisplayComponent({
...
+      component={ListItem}
+      emphasize={
+        JSON.stringify(optimizationTarget) ===
+        JSON.stringify([
+          sectionKey,
+          displayNs?.[key] ?? compareDisplayNs?.[key],
+        ])
+      }
+      diffOnly={sectionKey === 'character'}
+      showZero={sectionKey === 'baisc' && key === 'eleMas'}
```
Comment (nguyentvan7) [inline libs/gi/ui/src/components/character/StatDisplayComponent.tsx:112]:
```suggestion
      showZero={sectionKey === 'basic' && key === 'eleMas'}
```

Comment (nguyentvan7) [reply libs/gi/ui/src/components/character/StatDisplayComponent.tsx:112]: also prob want this for:
elemental dmg% of the character
phys dmg%
healing %

I think we used to have these overrides somewhere

File: libs/gi/db/src/Interfaces/BuildTc.ts (hunk 2 above)
Comment (frzyc) [reply libs/gi/db/src/Interfaces/BuildTc.ts:21]: Screenshot 2024-05-22 013432

Comment (frzyc) [reply libs/gi/db/src/Interfaces/BuildTc.ts:21]: Screenshot 2024-05-22 013424

**Review by nguyentvan7 — APPROVED** [review]

**Pull request merged**